integrations/cohere/src/haystack_integrations/components/generators/cohere/chat/chat_generator.py

```diff
@@ -202,11 +203,20 @@ def _parse_response(chat_response: ChatResponse, model: str) -> ChatMessage:
                     )
                 )
 
+        # Extract reasoning from content if present, even with tool calls
+        reasoning_content = None
+        if chat_response.message.content and hasattr(chat_response.message.content[0], "text"):
+            raw_content = chat_response.message.content[0].text
+            reasoning_content, _ = _extract_reasoning_from_response(raw_content)
```

Comment:
Re my comment [here](https://github.com/deepset-ai/haystack-core-integrations/pull/2266/files#r2526491898) it seems like we should be accessing `chat_response.message.content[0].thinking` to access thinking content.